components/os/component_output.go

```diff
@@ -436,7 +436,9 @@ func getDefaultPoller() query.Poller {
 	return defaultPoller
 }
 
-func CreateGet(cfg Config, eventsStore events_db.Store) func(ctx context.Context) (_ any, e error) {
+type getSystemdDetectVirtFunc func(ctx context.Context) (pkg_host.VirtualizationEnvironment, error)
+
+func createGet(cfg Config, eventsStore events_db.Store, getSystemdDetectVirt getSystemdDetectVirtFunc) func(ctx context.Context) (_ any, e error) {
```

Comment:
We don't need to refactor the function prototype.  If we want to test it, we can do as following:
```
var getSystemdDetectVirtFunc = pkg_host.SystemdDetectVirt

func createGet(.........) {
   // .... something
  virtEnv, err := getSystemdDetectVirtFunc()
  // blablabla....
}

func Test_createGet(t *testing.T) {
    defer func() {
       getSystemdDetectVirtFunc = pkg_host.SystemdDetectVirt
    }() 
   getSystemdDetectVirtFunc = func() (bool, error) {
        return false, context.DeadlineExceeded
    }
}

```

Reply:
done